### ISSUE TITLE
unit test pkg/kubelet/oom TestStartingWatcher fails when run in an unprivileged container

### ISSUE THREAD

**Issue by ixdy** [kubernetes/kubernetes#90883]
**What happened**: when one attempts to run the tests in pkg/kubelet/oom inside an unprivileged container, the `TestStartingWatcher` test case fails because `/dev/kmsg` doesn't exist:

```
--- FAIL: TestStartingWatcher (0.00s)                                                                                                             
    oom_watcher_linux_test.go:48:                                                                                                                 
                Error Trace:    oom_watcher_linux_test.go:48                                                                                      
                Error:          Received unexpected error:                                                                                        
                                open /dev/kmsg: no such file or directory                                                                         
                Test:           TestStartingWatcher                                                                                               
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xc1bc97]

goroutine 8 [running]:
testing.tRunner.func1(0xc0001a6300)
        /usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0xcca9a0, 0x14e3340)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/pkg/kubelet/oom.TestStartingWatcher(0xc0001a6300)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/oom/oom_watcher_linux_test.go:49 +0xc7
testing.tRunner(0xc0001a6300, 0xe082c0)
        /usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:960 +0x350
FAIL    k8s.io/kubernetes/pkg/kubelet/oom       0.101s
FAIL
```

**How to reproduce it (as minimally and precisely as possible)**:
The easiest way to run it locally is to use `build/run.sh` combined with make:
```
build/run.sh make test WHAT=./pkg/kubelet/oom
```

I noticed this test failing on the experimental **ci-kubernetes-cached-make-test** job (e.g. [this run](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-cached-make-test/1258602578921394176)), which runs in an unprivileged container.

This test failed under the bazel job for a while, but was [mitigated](https://github.com/kubernetes/kubernetes/pull/77046) by promoting that job to use a privileged container.

There was some discussion around how to fix this test properly in https://github.com/kubernetes/kubernetes/pull/77441, but progress appears to have been lost.

**Comment by ixdy** [comment]
It looks like https://github.com/kubernetes/kubernetes/pull/89580 might fix this... cc @rphillips

**Comment by ixdy** [comment]
this should be fixed by #90927.

cc @BenTheElder

**Comment by BenTheElder** [comment]
fixed https://github.com/kubernetes/kubernetes/pull/90927
